structuredheadings/plugin.js

```diff
@@ -464,6 +466,113 @@
+                editor.applyStyle(new CKEDITOR.style({
+                  element: editor.config.numberedElements[level]
+                }));
+
+                // Apply numbering also
+                editor.execCommand("autoNumberHeading");
+
+                return true;
+
+              /* if another list is found, and deep is enabled, recursively call the worker
+               * and incremement the level, deep will convert the entire list
+               */
+              } else if (node.type === CKEDITOR.NODE_ELEMENT && node.is({ul: 1, ol: 1}) && deep) {
+                processList(node, level + 1);
+                return false;
+              } else {
+                return false;
+              }
+            }, null, true);
+          };
+
+          //get the current list root element from the selection
+          var listRoot = editor.elementPath().contains(["ol", "ul"], false, true);
+          var deep = false; //don't deep process
+
+          if (listRoot) {
+
+            var previousHeader = getPreviousHeader(editor, listRoot);
+            if (!previousHeader) {
+                //fire off the worker at level 0
+              processList(listRoot, 0, deep);
+            } else {
+                //fire off worker at previous + 1
+              processList(listRoot,
+                    editor.config.numberedElements.indexOf(previousHeader.getName()) + 1,
+                    deep
+                );
+            }
+
+              /* After the process is run, select the entire list we worked on
+               * and outdent it, pushing the top level items out of the list. if
+               * deep is enabled, we instead flatten the entire list.
+               */
+            selectArea(listRoot, listRoot, true);
+
+            if (!deep) {
+              editor.execCommand("outdent");
+            } else {
+              editor.execCommand("bulletedlist");
+            }
+
+            /*
+             * At this point, everything has worked but the entire list is
+             * visibly selected, the original selection is invalid because the
+             * element has changed, bookmarks don't work for the same reason
+             * my only solution was to place the cursor at the start or end of
+             * the selection (first or last header created) assuming one would
+             * want to check the list, I chose start.
+             */
+
+            //end with selection at the first list element to cleanup
+            selectArea(
+                editor.getSelection().getStartElement(),
+                editor.getSelection().getStartElement()
+            );
+            //end with selection at the last list element
+//            selectArea(
+//                editor.getSelection().getRanges()[0].getTouchedEndNode(),
+//                editor.getSelection().getRanges()[0].getTouchedEndNode()
+//            );
+//            editor.getSelection().scrollIntoView();
```

Comment:
Depending on if Wes prefers to have the new selection at the first or last element of the converted list. commented and sitting, pending UX review.